### PR TITLE
Change Group ID to "com.paiondata.elide"

### LINKED ISSUE #4: Change Group ID to "com.paiondata.elide"
>[!TIP]
> As a maintainer of [an open source project](https://github.com/paion-data/elide), @Doom9527 should learn how to do large refactoring. This enhances both your programming skill and project management skill. This issue helps you grow in that skill

Goal of This Issue
------------------

**Change project Group ID to `com.paiondata.elide` so that we could *quickly* iterate Elide in our own forks**

[Our experience](https://github.com/yahoo/elide/pull/3191) shows that lots of Open Source communities aren't able to respond to contributors in a timely manner, probably due to their internal workloads, which is fairly acceptable.

As a tech startup that evolves much quicker, we should fork Elide and quickly iterate it in our own organization at a much faster pace. To do that, we must *[publish our own Elide fork to Maven central](https://github.com/orgs/paion-data/projects/10/views/1?pane=issue&itemId=57675835)*. The prerequisite of it, however, is to first change the [Group ID](https://maven.apache.org/guides/getting-started/#:~:text=groupId%20This%20element%20indicates%20the,maven.) of our fork, because without it we would not be able to publish our fork to Maven Central due to [namespace](https://central.sonatype.org/register/namespace/) collision

How to Implement
----------------

- Change [all `com.yahoo.elide`](https://github.com/search?q=repo%3Apaion-data%2Felide+com.yahoo.elide&type=code&p=1) to `com.paiondata.elide`

> [!TIP]
> - There will be lots of lines of code changes.
> - Therefore, using [IntelliJ's automatic "refactor" feature](https://www.jetbrains.com/help/idea/refactoring-source-code.html#refactoring_invoke) to do batch renaming would be a good idea

Edit: @Doom9527 suggested a very good point: **[Change version to 1.0.0](https://softwareengineering.stackexchange.com/a/31557)**

### PR DESCRIPTION
**Pull request by Doom9527**
Resolves #4

## Description
This is a parallel branch to dev03, and I tried to fix the bug using a different approach.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by QubitPi — CHANGES_REQUESTED** [review]

File: elide-core/src/main/java/com/paiondata/elide/core/PersistentResource.java [hunk 1]
```diff
@@ -203,7 +203,8 @@ public static <T> PersistentResource<T> createObject(
         requestScope.getNewPersistentResources().add(newResource);
         checkPermission(CreatePermission.class, newResource);
 
-        newResource.auditClass(Audit.Action.CREATE, new ChangeSpec(newResource, null, null, newResource.getObject()));
+        newResource.auditClass(Audit.Action.CREATE, new ChangeSpec(newResource, null, null,
+                newResource.getObject()));
```
[the remark this answers is not in the document]
Comment (Doom9527) [reply elide-core/src/main/java/com/paiondata/elide/core/PersistentResource.java:207]: 代码长度超过120

File: elide-core/src/main/java/com/paiondata/elide/core/PersistentResource.java [hunk 2]
```diff
@@ -648,7 +649,8 @@ private Object copyComplexAttribute(Object object) {
         EntityBinding binding = dictionary.getEntityBinding(type);
 
         Preconditions.checkState(! binding.equals(EMPTY_BINDING), "Model not found.");
-        Preconditions.checkState(binding.apiRelationships.isEmpty(), "Deep copy of relationships not supported");
+        Preconditions.checkState(binding.apiRelationships.isEmpty(),
+                "Deep copy of relationships not supported");
```
[the remark this answers is not in the document]
Comment (Doom9527) [reply elide-core/src/main/java/com/paiondata/elide/core/PersistentResource.java:653]: 代码长度超过120

File: elide-core/src/main/java/com/paiondata/elide/core/PersistentResource.java [hunk 3]
```diff
@@ -882,7 +884,8 @@ public boolean clearRelation(String relationName) {
                         .forEach(toDelete -> {
                             deletedRelationships.add(toDelete.getObject());
                         });
-                modifyCollection(collection, relationName, Collections.emptySet(), deletedRelationships, true);
+                modifyCollection(collection, relationName, Collections.emptySet(),
+                        deletedRelationships, true);
```
[the remark this answers is not in the document]
Comment (Doom9527) [reply elide-core/src/main/java/com/paiondata/elide/core/PersistentResource.java:888]: 代码长度超过120

File: elide-core/src/main/java/com/paiondata/elide/ElideModules.java [hunk 4]
```diff
@@ -17,13 +17,16 @@ private ElideModules() {
...
 
     static {
-        GRAPHQL_PRESENT = isPresent("com.yahoo.elide.graphql.GraphQLSettings", ElideModules.class.getClassLoader());
-        JSON_API_PRESENT = isPresent("com.yahoo.elide.jsonapi.JsonApiSettings", ElideModules.class.getClassLoader());
-        ASYNC_PRESENT = isPresent("com.yahoo.elide.async.AsyncSettings", ElideModules.class.getClassLoader());
+        GRAPHQL_PRESENT = isPresent("com.paiondata.elide.graphql.GraphQLSettings",
+                ElideModules.class.getClassLoader());
+        JSON_API_PRESENT = isPresent("com.paiondata.elide.jsonapi.JsonApiSettings",
+                ElideModules.class.getClassLoader());
+        ASYNC_PRESENT = isPresent("com.paiondata.elide.async.AsyncSettings",
```
[the remark this answers is not in the document]
Comment (Doom9527) [reply elide-core/src/main/java/com/paiondata/elide/ElideModules.java]: 好的

**Review by QubitPi — APPROVED** [review]

**Pull request merged**